swig/python2/tests/SysrepoBasicTest.py

```diff
@@ -0,0 +1,168 @@
+    elif (value.type() == sr.SR_INT8_T):
+        print "= " + repr(value.data().get_int8())
+    elif (value.type() == sr.SR_INT16_T):
+        print "= " + repr(value.data().get_int16())
+    elif (value.type() == sr.SR_INT32_T):
+        print "= " + repr(value.data().get_int32())
+    elif (value.type() == sr.SR_INT64_T):
+        print "= " + repr(value.data().get_int64())
+    elif (value.type() == sr.SR_IDENTITYREF_T):
+        print "= " + repr(value.data().get_identityref())
+    elif (value.type() == sr.SR_BITS_T):
+        print "= " + repr(value.data().get_bits())
+    elif (value.type() == sr.SR_BINARY_T):
+        print "= " + repr(value.data().get_binary())
+    else:
+        print "(unprintable)"
+
+
+class SysrepoBasicTest(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(self):
+        TestModule.create_test_module()
+        self.conn= sr.Connection("abc", sr.SR_DS_STARTUP)
+
+    def setUp(self):
+        TestModule.create_test_module()
+        self.session = sr.Session(self.conn, sr.SR_DS_STARTUP)
+
+    # No point in test?
+    # def test_connection(self):
+    #     with self.assertRaises(RuntimeError):
+    #         broken = sr.Connection('Reuqire daemon', 1)
+
+    def test_logg_stderr(self):
+        # sr.Connection.set_stderr(sr.SR_LL_DBG)
+        log = sr.Logs()
+        log.set_stderr(sr.SR_LL_NONE)
+
+    def test_get_item(self):
+        item = self.session.get_item("/test-module:main/i32")
+        print_value(item)
+        self.assertEqual(item.xpath(), "/test-module:main/i32")
+        self.assertEqual(item.type(), sr.SR_INT32_T)
+
+    def test_list_schema(self):
+        schemas = self.session.list_schemas()
+
+    def test_get_items(self):
+        vals = self.session.get_items("/test-module:main")
+        for i in xrange(vals.val_cnt()):
+            v = vals.val(i)
+            self.assertRegexpMatches(v.xpath(), "/test-module:main*")
+            print_value(v)
+
+    # Infinite loop on 'None' values
+    def test_get_items_iter(self):
+        it = self.session.get_items_iter("/test-module:main//*")
+        while True:
+            val = self.session.get_item_next(it)
+            if val is None: break
+            self.assertRegexpMatches(val.xpath(), "/test-module:main*")
+
+    def test_set_item(self):
+        xpath = "/example-module:container/list[key1='abc'][key2='def']/leaf"
+        v = sr.Val("Hey hou", sr.SR_STRING_T)
+        self.session.set_item(xpath, v)
+
+        new_value = self.session.get_item(xpath)
+        self.assertEqual(new_value.type(), sr.SR_STRING_T)
+        self.assertEqual(new_value.data().get_string(), v.data().get_string())
+
+    # This test used to put values to None in loop below. At the moment that segfaults.
+    # It also did not add 'type' and 'full' name to xpath.
+    def test_delete_item(self):
+        lists = ["/test-module:user[name='A']", "/test-module:user[name='B']", "/test-module:user[name='C']" ]
+        v = sr.Val("", sr.SR_STRING_T)
```

Comment:
There was an error in bindings regarding setting to None but it was fixed some time ago and somehow I missed updating this test. Thanks.